### PR TITLE
[FAB-18318] IT: verify DUPLICATE_TXID error when resubmitting old tx on a peer bootstrapped from snapshot

### PR DESCRIPTION
**Pull request by wenjianqiao**
Signed-off-by: Wenjian Qiao <wenjianq@gmail.com>

#### Type of change
- New feature

#### Description
Add test to verify verify DUPLICATE_TXID error when submitting old transactions on a peer bootstrapped from snapshot.

#### Related issues
https://jira.hyperledger.org/browse/FAB-18318

### CODE REVIEW

File: integration/ledger/snapshot_test.go [hunk 1]
```diff
@@ -1005,3 +1028,100 @@ func invokeWithoutPassingOrdererEndPoint(n *nwo.Network, peer *nwo.Peer, channel
...
+	broadcastClient := &common.BroadcastGRPCClient{Client: bc}
+	err = broadcastClient.Send(env)
+	Expect(err).NotTo(HaveOccurred())
+
+	// wait for deliver event
+	By("waiting for deliver event on peer " + peer.ID())
+	return dg.Wait(ctx)
+}
+
+func loadLocalMSPAt(dir, id, mspType string) (msp.MSP, error) {
```
Comment (yacovm) [inline integration/ledger/snapshot_test.go]: Here is an [easier way](https://github.com/hyperledger/fabric/blob/master/integration/raft/cft_test.go#L895-L901) to create a signer than to make a local MSP.

Comment (wenjianqiao) [reply integration/ledger/snapshot_test.go]: Thank you Yacov. Done.

File: integration/ledger/snapshot_test.go [hunk 2]
```diff
@@ -1005,3 +1037,68 @@ func invokeWithoutPassingOrdererEndPoint(n *nwo.Network, peer *nwo.Peer, channel
 	invokeChaincode(n, peer, command)
 	nwo.WaitUntilEqualLedgerHeight(n, channelID, nwo.GetLedgerHeight(n, peer, channelID), n.PeersWithChannel(channelID)...)
 }
+
+// commitTx commits a transaction for a given transaction envelope
+func commitTx(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channelID string, env *cb.Envelope, txid string) error {
```
Comment (manish-sethi) [inline integration/ledger/snapshot_test.go:1042]: In this function below, is there a problem if you simply wait for ledger height to increase by one as done in other functions in this file?

Comment (manish-sethi) [reply integration/ledger/snapshot_test.go:1042]: Got it... then you would have to do more work to get the validation code from block.

**Review by manish-sethi — APPROVED** [review]

**Pull request merged**